// admin-bro/lib/frontend/components/app/records-table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _recordInList = _interopRequireDefault(require("./record-in-list"));

var _propertyHeader = _interopRequireDefault(require("./property-header"));

var _viewHelpers = _interopRequireDefault(require("../../../backend/utils/view-helpers"));

var _types = require("../../types");

var _table = _interopRequireDefault(require("../ui/table"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const RecordsTable = props => {
  const {
    resource,
    paths,
    records,
    actionPerformed
  } = props;
  const h = new _viewHelpers.default();
  const newAction = h.resourceActionUrl({
    resourceId: resource.id,
    actionName: 'new'
  });

  if (!records.length) {
    return _react.default.createElement("div", {
      className: "content has-text-centered"
    }, _react.default.createElement("h3", null, "No records"), _react.default.createElement("p", null, "There are no records in this resource. Create", ' ', _react.default.createElement(_reactRouterDom.Link, {
      to: newAction
    }, "first record")));
  }

  return _react.default.createElement(_table.default, null, _react.default.createElement("thead", null, _react.default.createElement("tr", {
    key: "header"
  }, resource.listProperties.map(property => _react.default.createElement(_propertyHeader.default, {
    resource: resource,
    property: property,
    key: property.name
  })), _react.default.createElement("th", {
    key: "actions",
    style: {
      width: 80
    }
  }))), _react.default.createElement("tbody", null, records.map(record => _react.default.createElement(_recordInList.default, {
    record: record,
    resource: resource,
    paths: paths,
    key: record.id,
    actionPerformed: actionPerformed
  }))));
};

RecordsTable.propTypes = {
  resource: _types.resourceType.isRequired,
  paths: _types.pathsType.isRequired,
  records: _propTypes.default.arrayOf(_types.recordType).isRequired,
  actionPerformed: _propTypes.default.func.isRequired
};
var _default = RecordsTable;
exports.default = _default;
